### DISCUSSION TITLE
Allow creating workspace above top workspace

### DISCUSSION THREAD

**Discussion by Supreeeme** [YaLTeR/niri#247] [Ideas]
Currently, when you're on the topmost workspace, if you want to move a single window above, you have to manually move all the other windows on the workspace to a workspace below, and this could be problematic if you already had a workspace below that you wanted to keep separate. If it was allowed to send a window to a new workspace above the topmost workspace, this would be alleviated.

**Comment by YaLTeR** [comment]
Maybe an action to "switch to a new empty workspace right below/above"?

**Reply by Supreeeme** [reply]
I was originally thinking to just extend the focus workspace above action to create a new workspace like the focus workspace below does. But, I think a more general action to create an empty workspace above/below seems like it could have more utility and let you insert new workspaces in between existing ones. Sounds good to me.

**Reply by keks** [reply]
> Maybe an action to "switch to a new empty workspace right below/above"?

I was just about to start a discussion to ask for that feature! I think that would be super helpful, together with "move current window to a new workspace below/above the current"

**Reply by YaLTeR** [reply]
Specifically for above top, in 25.01 there's a new option: https://github.com/YaLTeR/niri/wiki/Configuration:-Layout#empty-workspace-above-first

**Reply by klardotsh** [reply]
"Switch to a new empty workspace directly above/below current" is a feature I would absolutely love. I spent today getting familiar with Niri and mostly love it, but right now I'm sitting on a "crap, I really want to make a workspace for this project, but to put the workspace in the order I want, I'd have to manually move dozens of windows on the subsequent workspaces" problem, and haven't found a workaround yet.

**Reply by YaLTeR** [reply]
I usually go all the way down, open a new window, and then `move-workspace-up` it to where I want.

**Reply by marcinjahn** [reply]
> Specifically for above top, in 25.01 there's a new option: https://github.com/YaLTeR/niri/wiki/Configuration:-Layout#empty-workspace-above-first

This feature is great, but I think it'd be even better if the new workspace would appear only when it's needed. What I mean is that `empty-workspace-above-first` creates an empty workspace in the top that is always there. I think it'd be nice if the workspace could be created dynamically when you actually try to move some window UP (and you're already in the first workspace). Currently, when I enable the option, the empty workspace is not needed most of the time, and when I want to go to my first workspace by going up from another workspace down below, I always hit the empty first workspace when I actually wanted to get to the second one (which is populated with windows).
Currently, first empty workspace behaves pretty much exactly the same as the last free workspace. And I think that's not correct. The last one should always be there, like it's now, but the first one should appear only when it's requested. Potentially, the other feature discussed where "Switch to a new empty workspace directly above/below current" could be it.

**Reply by YaLTeR** [reply]
fwiw the new overview lets you drag with a mouse to a new workspace above/between

**Reply by marcinjahn** [reply]
Yeah, I'm awaiting this feature to be available in the main package on COPR, it's awesome! Still, keyboard-only flows are important as well :slightly_smiling_face:
